src/amdgpu.cpp

```diff
@@ -88,23 +89,87 @@ void amdgpu_get_instant_metrics(struct amdgpu_common_metrics *metrics) {
 		metrics->gpu_load_percent = amdgpu_metrics->average_gfx_activity;
 
 		metrics->average_gfx_power_w = amdgpu_metrics->average_gfx_power / 1000.f;
-		metrics->average_cpu_power_w = amdgpu_metrics->average_cpu_power / 1000.f;
-		// Use cpu + soc power similar to 'get_cpu_power_k10temp' in 'cpu.cpp' ?
-		//metrics->average_cpu_power_w = amdgpu_metrics->average_cpu_power / 1000.f + amdgpu_metrics->average_soc_power / 1000.f;
 
-		// average_cpu_power_w fallback ?
-		//if(!( amdgpu_metrics->average_cpu_power ^ 0xffff && amdgpu_metrics->average_soc_power ^ 0xffff ))
-		//    metrics->average_cpu_power_w = amdgpu_metrics->average_socket_power / 1000.f - amdgpu_metrics->average_gfx_power / 1000.f;
+		if( IS_VALID_METRIC(amdgpu_metrics->average_cpu_power) && IS_VALID_METRIC(amdgpu_metrics->average_soc_power) ) {
+			// Use 'cpu power' + 'soc power' similar to 'get_cpu_power_k10temp' in 'cpu.cpp'
+			metrics->average_cpu_power_w = amdgpu_metrics->average_cpu_power / 1000.f + amdgpu_metrics->average_soc_power / 1000.f;
```

Comment:
Why would cpu_power be cpu_power + soc_power ?

Reply:
From how I understand it `cpu_power` is the cpu core power while `soc_power` is the soc power. To make the reported power more consistent with what is reported by the function `get_cpu_power_k10temp` in the file `cpu.cpp`, I used cpu core power + soc power.

The function `get_cpu_power_k10temp` in the file `cpu.cpp` uses the following code:
```
power = (coreVoltage * coreCurrent + socVoltage * socCurrent) / 1000000;
```

I tried to look up the meaning of `cpu_power` again just now but could not find additional information on it. I don't know anymore where I got the information/idea that `cpu_power` is only the cpu core power, so I could be wrong. If you don't like it, then I can always change it to only use `cpu_power`.

Reply:
Sorry for the late replay, the comment was "PENDING" and I didn't notice it.

Reply:
Do you want me to change the code to the following?

```
metrics->average_cpu_power_w = amdgpu_metrics->average_cpu_power / 1000.f;
```

Reply:
Yes I think that should be better

Reply:
oké, I will change it then.